File: src/main/java/com/blink/backend/controller/message/dto/AckStatus.java
package com.blink.backend.controller.message.dto;

import lombok.Getter;

@Getter
public enum AckStatus {
    ERROR("ERROR"),
    PENDING("PENDING"),
    SERVER("SENT"),
    DEVICE("RECEIVED"),
    READ("READ"),
    PLAYED("PLAYED"),
    UNKNOWN("UNKNOWN");

    /*ERROR,
    PENDING,
    SERVER,
    DEVICE,
    READ,
    PLAYED,
    UNKNOWN;*/

    private final String value;

    AckStatus(String value) {
        this.value = value;
    }
}

File: src/main/java/com/blink/backend/domain/integration/waha/dto/AckStatus.kt
package com.blink.backend.domain.integration.waha.dto

import com.blink.backend.domain.chat.model.WhatsAppAckStatus

enum class AckStatus {
    ERROR,
    PENDING,
    SERVER,
    DEVICE,
    READ,
    PLAYED,
    UNKNOWN,
    ;

    fun toDomain(): WhatsAppAckStatus {
        return when (this) {
            PENDING ->  WhatsAppAckStatus.PENDING
            SERVER ->  WhatsAppAckStatus.SENT
            DEVICE ->  WhatsAppAckStatus.RECEIVED
            READ ->  WhatsAppAckStatus.READ
            else ->  WhatsAppAckStatus.OTHER
        }
    }
}